src/stringParamAllocator.js

```diff
@@ -0,0 +1,92 @@
+import { isArrayAllStrings } from './validation.js';
+
+/* global Module */
+export default class ParamParser {
```

Comment:
I like your perspective on using an object oriented approach here. I agree, the internalization of how we handle Loam logic is less relevant to the end users of the API. I think it will be fine to maintain the parameter parsing logic within an internally instantiated class instance.